### PR TITLE
Only change configurations locally from old V1 to new V2.

### PR DESCRIPTION
**Pull request by natsegal**

### CODE REVIEW

File: test/test_framework.py [hunk 1]
```diff
@@ -314,9 +314,10 @@ def test_train(tmpdir):
     model_dir_old = _run_framework(tmpdir, "model1", "train", config=config_base_old)
     config = _read_config(model_dir_old)
     assert os.path.isfile(
-        os.path.join(model_dir_old, os.path.basename(config["vocabulary"]["source"]["path"])))
+        os.path.join(model_dir_old, os.path.basename(config["tokenization"]["source"]["vocabulary"])))
     assert os.path.isfile(
-        os.path.join(model_dir_old, os.path.basename(config["vocabulary"]["target"]["path"])))
+        os.path.join(model_dir_old, os.path.basename(config["tokenization"]["target"]["vocabulary"])))
```
Comment (guillaumekln) [inline test/test_framework.py:319]: Can you also check in this test that the `vocabulary` field does not exist?

File: nmtwizard/framework.py [hunk 2]
```diff
@@ -842,29 +850,43 @@ def _get_vocab_info(self,
         if previous_vocab is not None:
             previous_vocab = self._convert_vocab(
                 previous_vocab, basename=previous_basename)
-            del opt['previous_vocabulary']
+            if 'previous_vocabulary' in opt:
+                del opt['previous_vocabulary']
```
Comment (guillaumekln) [inline nmtwizard/framework.py:854]: You could replace all `if x in y: del y[x]` with `y.pop(x, None)` to make the code more compact.

File: nmtwizard/preprocess/tokenizer.py [hunk 3]
```diff
@@ -11,6 +11,7 @@ def build_tokenizer(args):
     """Builds a tokenizer based on user arguments."""
     args = args.copy()
     args.pop('vocabulary', None)
+    args.pop('replace_vocab', None)
```
Comment (guillaumekln) [inline nmtwizard/preprocess/tokenizer.py:14]: Is this one needed even though `_get_vocab_info` is removing it?

**Review by guillaumekln — CHANGES_REQUESTED** [review]

File: nmtwizard/preprocess/tokenizer.py (hunk 3 above)
Comment (natsegal) [reply nmtwizard/preprocess/tokenizer.py:14]: To my understanding, _get_vocab_info is activated after _build_data which uses the original tokenization field.

In new configurations 'replace_vocab' option is in a separate vocabulary field, so no problem.
In old configurations, however, it is supposed to be in 'tokenization' field, so I get an error in 'test_replace_vocab_old' test if I remove it.

**Review by guillaumekln — APPROVED** [review]

**Pull request merged**